botocore/docs/sharedexample.py

```diff
@@ -0,0 +1,166 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import re
+from botocore.vendored.six import string_types
+
+
+class SharedExampleDocumenter(object):
+    def document_shared_example(self, example, prefix, section):
+        """Documents a single shared example based on its definition.
+
+        :param example: The model of the example
+
+        :param prefix: The prefix to use in the method example.
+
+        :param section: The section to write to.
+        """
+        section.style.new_paragraph()
+        section.write(example.get('description'))
+        section.style.new_line()
+        self.document_input(section, example, prefix)
+        self.document_output(section, example)
+
+    def document_input(self, section, example, prefix):
+        input_section = section.add_new_section('input')
+        input_section.style.start_codeblock()
+        if prefix is not None:
+            input_section.write(prefix)
+        params = example['input']
+        comments = example.get('comments')
+        if comments:
+            comments = comments.get('input')
+        param_section = input_section.add_new_section('parameters')
+        self._document_params(param_section, params, comments, [])
+        closing_section = input_section.add_new_section('input-close')
+        closing_section.style.new_line()
+        closing_section.style.new_line()
+        closing_section.write('print(response)')
+        closing_section.style.end_codeblock()
+
+    def document_output(self, section, example):
+        output_section = section.add_new_section('output')
+        output_section.writeln('Expected Output:')
+        output_section.style.start_codeblock()
+
+        if 'output' in example:
+            params = example['output']
+        else:
+            params = {}
+
+        # There might not be an output, but we will return metadata anyway
+        params['ResponseMetadata'] = {"...": "..."}
+        comments = example.get('comments')
+        if comments:
+            comments = comments.get('output')
+        self._document_dict(output_section, params, comments, [], True)
+        closing_section = output_section.add_new_section('output-close')
+        closing_section.style.end_codeblock()
+
+    def _document(self, section, value, comments, path):
+        if isinstance(value, dict):
+            self._document_dict(section, value, comments, path)
+        if isinstance(value, list):
+            self._document_list(section, value, comments, path)
+        if isinstance(value, string_types):
```

Comment:
What if the value is not a string (like an integer or datetime). How does that get handled?